crates/pet/src/find.rs

```diff
@@ -53,6 +54,7 @@ pub fn find_and_report_envs(
     let conda_executable = configuration.conda_executable;
     thread::scope(|s| {
         // 1. Find using known global locators.
+        let os_env = os_environment.clone();
```

Comment:
If the goal is to avoid re-parsing the environment in paths like these, I would have them take an owned `dyn Environment` (when needed) and internally construct an `Arc` for it to give as a normal borrowed `&dyn Environment` to consumers